cdm/core/src/main/java/ucar/nc2/internal/iosp/hdf4/H4iosp.java

```diff
@@ -475,14 +492,36 @@ public ByteBuffer getByteBuffer() throws IOException {
   @Override
   public void reacquire() throws IOException {
     super.reacquire();
-    header.raf = this.raf;
+    getHeader().raf = this.raf;
   }
 
   public Object sendIospMessage(Object message) {
-    if (message.toString().equals("header"))
-      return header;
+    if (message instanceof Charset) {
+      setValueCharset((Charset) message);
+    }
+    if (message.toString().equals("header")) {
+      return getHeader();
+    }
     return super.sendIospMessage(message);
   }
 
+  /**
+   * Return {@link Charset value charset} if it was defined. Definition of charset
+   * occurs by sending a charset as a message using the {@link #sendIospMessage}
+   * method.
+   * 
+   * @return {@link Charset value charset} if it was defined.
+   */
+  protected Optional<Charset> getValueCharset() {
+    return Optional.ofNullable(valueCharset);
+  }
 
+  /**
+   * Define {@link Charset value charset}.
+   * 
+   * @param charset may be null.
+   */
+  protected void setValueCharset(Charset charset) {
```

Comment:
Would you mind annotating this method's signature with `@Nullable` from `javax.annotation`, so something like `setValueCharset(@Nullable Charset charset)`?